spec/models/pacbio/well_spec.rb

```diff
@@ -317,5 +324,77 @@
         end
       end
     end
+
+    context 'v12_revio' do
+      # build the well with the smrt_link version 12 revio
+      let(:well) { build(:pacbio_well, plate: create(:pacbio_plate, run: create(:pacbio_run, smrt_link_version: version12_revio))) }
+
+      # before do - create all the version12_revio specific options in here
+      before do
+        create(:pacbio_smrt_link_option, key: 'movie_acquisition_time', validations: { presence: {}, numericality: { greater_than_or_equal_to: 0.1, less_than_or_equal_to: 30 } }, smrt_link_versions: [version12_revio])
+        create(:pacbio_smrt_link_option, key: 'include_base_kinetics', validations: { presence: {}, inclusion: { in: true_false } }, smrt_link_versions: [version12_revio])
+        create(:pacbio_smrt_link_option, key: 'library_concentration', validations: { presence: {} }, smrt_link_versions: [version12_revio])
+        create(:pacbio_smrt_link_option, key: 'polymerase_kit', validations: { presence: {} }, smrt_link_versions: [version12_revio])
+      end
+
+      context 'Movie acquisition time' do
+        it 'must be present' do
+          well.movie_acquisition_time = nil
+          expect(well).not_to be_valid
+        end
+
+        it 'can be a decimal' do
+          well.movie_acquisition_time = 0.2
+          expect(well.movie_acquisition_time).to eq(0.2)
+        end
+
+        describe 'must be within range' do
+          it 'is within range' do
+            expect(well.movie_acquisition_time = 15).to eq(15)
+          end
+
+          it 'is above range' do
+            well.movie_acquisition_time = 31
+            expect(well).not_to be_valid
+          end
+
+          it 'is below range' do
+            well.movie_acquisition_time = 0
+            expect(well).not_to be_valid
+          end
+        end
+      end
+
+      context 'Include base kinetics' do
+        it 'must be present' do
+          well.include_base_kinetics = nil
+          expect(well).not_to be_valid
+        end
+
+        it 'must be a valid value' do
+          true_false.each do |option|
+            well.include_base_kinetics = option
+            expect(well).to be_valid
+          end
+
+          well.include_base_kinetics = 'junk'
+          expect(well).not_to be_valid
+        end
+      end
+
+      context 'Library concentration' do
+        it 'must have a library concentration' do
```

Comment:
Would add a test here and in the 'polymerase kit' tests to check it is valid when it has a valid value